05.wc/wc.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = parse_options
+  if ARGV.empty?
+    handle_stdin(options)
+  else
+    handle_files(ARGV, options)
+  end
+end
+
+def parse_options
+  options = { lines: false, words: false, bytes: false }
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: wc.rb [options] [file ...]'
+    opts.on('-l', 'Count lines') { options[:lines] = true }
+    opts.on('-w', 'Count words') { options[:words] = true }
+    opts.on('-c', 'Count bytes') { options[:bytes] = true }
+  end.parse!
+  options.values.any? ? options : options.transform_values { true }
+end
+
+def handle_stdin(options)
+  content = $stdin.read
+  file_info = { counts: count_all(content) }
+  output_results(file_info, options)
+end
+
+def handle_files(files, options)
+  counts = files.map do |file|
+    content = File.read(file)
+    { counts: count_all(content), filename: file }
+  end
+  counts.each { |file_info| output_results(file_info, options) }
```

Comment:
ループを回すなら

```ruby
複数形.each { |単数形| ... }
```

の形にしたいですね。
https://bootcamp.fjord.jp/pages/readable-variable-name#%E3%83%AB%E3%83%BC%E3%83%97%E5%87%A6%E7%90%86%E3%81%AE%E5%A0%B4%E5%90%88%E3%81%AF%E3%80%8C%E9%85%8D%E5%88%97%E3%81%AF%E8%A4%87%E6%95%B0%E5%BD%A2%E3%80%81%E3%81%9D%E3%81%AE%E4%B8%AD%E3%81%AE%E5%90%84%E8%A6%81%E7%B4%A0%E3%81%AF%E5%8D%98%E6%95%B0%E5%BD%A2%E3%80%8D%E3%81%8C%E3%82%BB%E3%82%AA%E3%83%AA%E3%83%BC

ただ、info (information)の複数形はないので、infosにはしない方がいいです。
https://qiita.com/jnchito/items/459d58ba652bf4763820#%E5%90%8D%E5%89%8D%E3%82%92%E4%BB%98%E3%81%91%E3%82%8B%E3%81%A8%E3%81%8D%E3%81%AF%E5%8F%AF%E7%AE%97%E5%90%8D%E8%A9%9E%E3%81%8B%E4%B8%8D%E5%8F%AF%E7%AE%97%E5%90%8D%E8%A9%9E%E3%81%8B%E3%82%92%E6%84%8F%E8%AD%98%E3%81%99%E3%82%8B

listには配列の意味もあるので、この場合

```suggestion
  file_info_list.each { |file_info| output_results(file_info, options) }
```

みたいにするといいかもしれません

Reply:
@JunichiIto 
ドキュメント共有ありがとうございます🙇‍♂️
可算名詞か不可算名詞かを気をつけて命名していきます。